harmony_netcdf_to_zarr/adapter.py

```diff
@@ -52,68 +56,72 @@ def __init__(self, message, catalog=None, config=None):
+            one-to-one operations, it is assumed that the `concatenate` query
+            parameter is False, and Harmony will invoke this backend service
+            once per input granule. Because of this, each backend invocation is
+            expected to produce a single Zarr output.
 
-        Parameters
-        ----------
-        item : pystac.Item
-            the item that should be converted
-        source : harmony.message.Source
-            the input source defining the variables, if any, to subset from the item
-
-        Returns
-        -------
-        pystac.Item
-            a STAC item containing the Zarr output
         """
-        result = item.clone()
-        result.assets = {}
-
-        # Create a temporary dir for processing we may do
         workdir = mkdtemp()
         try:
-            # Get the data file
-            asset = next(v for k, v in item.assets.items() if 'data' in (v.roles or []))
-            input_filename = download(
-                asset.href,
-                workdir,
-                logger=self.logger,
-                access_token=self.message.accessToken,
-                cfg=self.config
-            )
-
-            name = generate_output_filename(asset.href, ext='.zarr')
-            root = self.message.stagingLocation + name
-
-            try:
-                store = self.s3.get_mapper(root=root, check=False, create=True)
-                netcdf_to_zarr(input_filename, store)
-            except Exception as e:
-                # Print the real error and convert to user-facing error that's more digestible
-                self.logger.error(e, exc_info=1)
-                filename = asset.href.split('?')[0].rstrip('/').split('/')[-1]
-                raise ZarrException(f'Could not convert file to Zarr: {filename}') from e
-
-            # Update the STAC record
-            result.assets['data'] = Asset(root, title=name, media_type='application/x-zarr', roles=['data'])
-
-            # Return the STAC record
-            return result
+            items = list(self.catalog.get_items())
+            netcdf_urls = get_netcdf_urls(items)
+
+            local_file_paths = download_granules(netcdf_urls, workdir,
+                                                 self.message.accessToken,
+                                                 self.config, self.logger)
+
+            # Enable during DAS-1379:
+            # dimensions_mapping = DimensionsMapping(local_file_paths)
+
+            if len(local_file_paths) == 1:
+                output_name = generate_output_filename(netcdf_urls[0],
+                                                       ext='.zarr')
+            else:
+                # Mimicking PO.DAAC Concise: for many-to-one the file name is
+                # "<collection>_merged.zarr".
+                collection = self._get_item_source(items[0]).collection
+                output_name = f'{collection}_merged.zarr'
```

Comment:
I had to dig around for this in the [unit tests](https://github.com/nasa/harmony-service-lib-py/blob/main/tests/test_adapter_stac.py#L39) for `harmony-service-lib-py` to confirm. Looking at the unit tests there, it looks like you are right; the URL will likely be the collection concept ID, not the short name. ([Here's the class method](https://github.com/nasa/harmony-service-lib-py/blob/main/harmony/adapter.py#L220), where it's essentially looking for the parent item in the STAC catalogue).

Reply:
Short-Name is preferable for end-user.  Is this possible?

Reply:
I definitely agree that a short name is more end-user friendly! I'm not sure that the short name is available to us - this is making use of a method on the `BaseHarmonyAdapter` class, so outside of the scope of the service itself.

I guess the other thing worth noting is that this is what the PO.DAAC Concise service is doing, so we're being consistent with existing services. I definitely think this is a good question, and think the short name would be preferable if available. This might be a conversation to float up to a higher level, to see if the collection short name can be made available to the services somehow.

Reply:
Addendum (sorry, still mulling this over): I think, mainly, this all stems from what Harmony chooses to put in the input STAC catalogue sent to the backend service.

(An alternative might be to delve into the metadata of one of the granules to find a short name in the global attributes, but then we'd be inconsistent with Concise)